### PR TITLE
Fire callback also on pure closing of prompt dialog

### LINKED ISSUE #2578: OC.dialogs doesn't trigger callback if dialog is closed through "X" or ESC
### Steps to reproduce
1. Open a dialog using `OC.dialogs.confirm` and pass a callback function.
2. Close the dialog by clicking the `X` in the top-right corner or press ESC.

### Expected behaviour
The callback should be executed with `false` as argument (because the user didn't confirm the dialog). This would be necessary to perform some cleanup code.

### Actual behaviour
The callback is not executed.

### PR DESCRIPTION
**Pull request by MorrisJobke**
Fixes #2578 
cc @nextcloud/javascript

### CODE REVIEW

File: core/js/oc-dialogs.js [hunk 1]
```diff
@@ -124,12 +124,14 @@ var OCdialogs = {
 				modal = false;
 			}
 			$('body').append($dlg);
+			var callbackAlreadyFired = false;
```
Comment (ChristophWurst) [inline core/js/oc-dialogs.js]: you could wrap the callback with ``_.once(…)`` ;-) http://underscorejs.org/#once

Comment (MorrisJobke) [inline core/js/oc-dialogs.js]: Done 👍

**Comment by ChristophWurst** [comment]
LGTM

**Comment by fancycode** [comment]
lgtm, thanks

**Review by fancycode — APPROVED** [review]

**Review by LukasReschke — APPROVED** [review]

**Pull request merged**